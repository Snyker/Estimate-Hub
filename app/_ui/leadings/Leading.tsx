import React, {DetailedHTMLProps, HTMLAttributes, PropsWithChildren} from "react";

export const Leading1: React.FC<PropsWithChildren<DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>>> = ({children, className = '', ...props}) => {
    return (
        <h1 className={`text-4xl text-center my-4 ${className}`} {...props}>
            {children}
        </h1>
    );
};

export const Leading2: React.FC<PropsWithChildren<DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>>> = ({children, className = '', ...props}) => {
    return (
        <h2 className={`text-3xl text-center my-3.5 ${className}`} {...props}>
            {children}
        </h2>
    );
};

export const Leading3: React.FC<PropsWithChildren<DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>>> = ({children, className = '', ...props}) => {
    return (
        <h3 className={`text-2xl text-center my-3 ${className}`} {...props}>
            {children}
        </h3>
    );
};
