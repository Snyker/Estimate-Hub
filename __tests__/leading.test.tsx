import renderer from 'react-test-renderer';
import {Leading1, Leading2, Leading3} from "@/app/_ui/leadings/Leading";

describe('Leading1 Component', () => {
    it('renders children correctly', () => {
        const tree = renderer.create(<Leading1>Hello, World!</Leading1>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom class name', () => {
        const tree = renderer.create(<Leading1 className="custom-class">Custom Heading</Leading1>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies additional HTML attributes', () => {
        const tree = renderer.create(<Leading1 id="custom-id">Heading with custom ID</Leading1>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Leading2 Component', () => {
    it('renders children correctly', () => {
        const tree = renderer.create(<Leading2>Hello, World!</Leading2>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom class name', () => {
        const tree = renderer.create(<Leading2 className="custom-class">Custom Heading</Leading2>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies additional HTML attributes', () => {
        const tree = renderer.create(<Leading2 id="custom-id">Heading with custom ID</Leading2>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});


describe('Leading3 Component', () => {
    it('renders children correctly', () => {
        const tree = renderer.create(<Leading3>Hello, World!</Leading3>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom class name', () => {
        const tree = renderer.create(<Leading3 className="custom-class">Custom Heading</Leading3>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies additional HTML attributes', () => {
        const tree = renderer.create(<Leading3 id="custom-id">Heading with custom ID</Leading3>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
