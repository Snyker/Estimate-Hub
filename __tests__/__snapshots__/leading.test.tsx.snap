// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leading1 Component applies additional HTML attributes 1`] = `
<h1
  className="text-4xl text-center my-4 "
  id="custom-id"
>
  Heading with custom ID
</h1>
`;

exports[`Leading1 Component applies custom class name 1`] = `
<h1
  className="text-4xl text-center my-4 custom-class"
>
  Custom Heading
</h1>
`;

exports[`Leading1 Component renders children correctly 1`] = `
<h1
  className="text-4xl text-center my-4 "
>
  Hello, World!
</h1>
`;

exports[`Leading2 Component applies additional HTML attributes 1`] = `
<h2
  className="text-3xl text-center my-3.5 "
  id="custom-id"
>
  Heading with custom ID
</h2>
`;

exports[`Leading2 Component applies custom class name 1`] = `
<h2
  className="text-3xl text-center my-3.5 custom-class"
>
  Custom Heading
</h2>
`;

exports[`Leading2 Component renders children correctly 1`] = `
<h2
  className="text-3xl text-center my-3.5 "
>
  Hello, World!
</h2>
`;

exports[`Leading3 Component applies additional HTML attributes 1`] = `
<h3
  className="text-2xl text-center my-3 "
  id="custom-id"
>
  Heading with custom ID
</h3>
`;

exports[`Leading3 Component applies custom class name 1`] = `
<h3
  className="text-2xl text-center my-3 custom-class"
>
  Custom Heading
</h3>
`;

exports[`Leading3 Component renders children correctly 1`] = `
<h3
  className="text-2xl text-center my-3 "
>
  Hello, World!
</h3>
`;
